test/getGitWorkspaceState.test.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as fse from 'fs-extra';
+import { join } from 'path';
+import { Uri } from "vscode";
+import { parseError } from 'vscode-azureextensionui';
+import { getGitWorkspaceState, GitWorkspaceState, promptForDefaultBranch, tryGetDefaultBranch, verifyGitWorkspaceForCreation } from "../extension.bundle";
+import { createTestActionContext, testFolderPath, testUserInput } from "./global.test";
+
+suite('Workspace Configurations for SWA Creation', function (this: Mocha.Suite): void {
+    this.timeout(2 * 60 * 1000);
+    const context = createTestActionContext();
+    const testFolderUri: Uri = Uri.file(testFolderPath);
+
+    test('Empty workspace with no git repository', async () => {
+        const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, testFolderUri)
+        let message: string = '';
+        try {
```

Comment:
I would use a util method like this:
https://github.com/microsoft/vscode-azuretools/blob/main/ui/test/assertThrowsAsync.ts